### ISSUE TITLE
Simultaneous write and read

### ISSUE THREAD

**Issue by kolia** [JuliaAudio/PortAudio.jl#37]
Hi @ssfrr 

How would you recommend playing a sound while simultaneously recording?  (Not a duplex stream, I’m not trying to play what is being recorded, I’m trying to play a predefined sound)

Thanks

**Comment by mchitre** [comment]
@ssfrr is the `synced` keyword working?

```
julia> PortAudioStream(1, 1; synced=true) do stream
         write(stream, 0.01*randn(4800))
         read(stream, 0.1s)
       end
ERROR: MethodError: no method matching PortAudioStream(::PortAudio.PortAudioDevice, ::PortAudio.PortAudioDevice, ::Int64, ::Int64; synced=true)
Closest candidates are:
  PortAudioStream(::PortAudio.PortAudioDevice, ::PortAudio.PortAudioDevice, ::Any, ::Any; eltype, samplerate, latency, warn_xruns, recover_xruns)
at /Users/mandar/.julia/packages/PortAudio/JRDxf/src/PortAudio.jl:154 got unsupported keyword argument "synced"
```

**Comment by fhchl** [comment]
It seems like the `synced` kwarg has been removed in code, but not from `README.md`.

**Comment by rob-luke** [comment]
Thanks for noticing that detail @fhchl. If it's in the readme then it should work. Or a note should be added to the readme on how to achieve the same functionality now.

@bramtayl why was the synced keyword removed? And how should users ensure that the input and output ringbuffers are kept synchronized now?

Any time functionality is removed it should involve a code review, and we should be sure to add depreciation warnings and note it in the documentation and release note.

**Comment by rob-luke** [comment]
Ok I’m still digging in to this, but synced seems to have been removed in 9eb565e487f2fe78ccd00ef41406e9cb1611ce31 January 2020.

**Comment by rob-luke** [comment]
@ssfrr is there an alternative suggestion on how to achieve deterministic latency with `synced` removed? Or should we just remove it from the README?

**Comment by bramtayl** [comment]
I don't know why `synced` was removed, sorry

**Comment by ssfrr** [comment]
We lost `synced` when I switched from the callback implemented in C to using libportaudio's streaming interface. It made the code a lot simpler but we lost some low-level control and are relying on portaudio to manage the io buffers.

Previously the C callback would stop pushing/pulling from the write/read ringbuffers when either one of them was full/empty (respectively). That ensured that the ringbuffer state was always synchronized, but made things trickier to manage because the system would hang up if you tried to read from a stream without simultaneously writing to it, or vice-versa.

In general it was a headache and it seemed like the vast majority of users don't really care about deterministic round-trip latency, so I gave it up. This isn't great if you're using it to try to do impulse response measurements or something, but changes in the round-trip latency should only happen on under/overflow, so your measurement would be hosed anyways.
